peekingduck/pipeline/nodes/output/csv_writer.py

```diff
@@ -92,7 +97,7 @@ def _check_tracked_stats(self, inputs: Dict[str, Any]) -> None:
         self._stats_to_track = valid
         self._stats_checked = True
 
-    def __del__(self) -> None:
+    def _reset(self) -> None:
         del self.csv_logger
```

Comment:
Out of curiosity, why is it required to delete csv_logger?

Reply:
The reason is because file is closed on delete `self.csv_file.close()`. [Seen here](https://github.com/aimakerspace/PeekingDuck/blob/d9eecaa991c8cf919c756ea5cc4d8a1b3dfdc917/peekingduck/pipeline/nodes/output/utils/csvlogger.py#L60)